### PR TITLE
V2

### PR DESCRIPTION
**Pull request by msailes**
*Issue #, if available:*
You didn't ask for it, but it enables future work.

*Description of changes:*
I've refactored the project to use the AWS SDK for Java v2.

I doing this I have changed how the configuration of server side encryption is managed. I've implemented it in one way, but I'm sure there could be better ways. 

This isn't a final cut, I'm missing documentation and unit test coverage. 

I thought it best to get the ball rolling on what is quite a big PR.

Thanks,

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -36,15 +36,22 @@
     </developers>
```
Comment (aws-rizi) [inline pom.xml:37]: I think we need to update version of the lib as well:

software.amazon.payloadoffloading
payloadoffloading-common
1.0.0. ==> to new version

Comment (msailes) [reply pom.xml:37]: Would you want that to be 2.0.0?

Comment (adam-aws) [reply pom.xml:37]: Yes this should be version 2.0.0, should include README.md change as well
`<version>2.0.0</version>`

File: pom.xml [hunk 2]
```diff
@@ -36,15 +36,22 @@
...
 
     <properties>
-        <aws-java-sdk.version>1.11.817</aws-java-sdk.version>
+        <aws-java-sdk.version>2.13.64</aws-java-sdk.version>
+        <commons.logging.version>1.1.3</commons.logging.version>
```
Comment (aws-rizi) [inline pom.xml:40]: I believe we dont need this, Please correct me if I am wrong.

Comment (msailes) [reply pom.xml:40]: You are right.

            <groupId>software.amazon.awssdk</groupId>
            <artifactId>utils</artifactId>

includes slf4j.

File: pom.xml (hunk 1 above)
Comment (msailes) [reply pom.xml:37]: fixed

File: README.md [hunk 3]
```diff
@@ -23,7 +23,7 @@ You can download release builds through the [releases section of this](https://g
   <dependency>
     <groupId>software.amazon.payloadoffloading</groupId>
     <artifactId>payloadoffloading-common</artifactId>
-    <version>1.0.0</version>
+    <version>2.0.0</version>
```
Comment (adam-aws) [inline README.md:26]: Sorry about that, but can we also leave the v1, so we would have something like
### Version 2.x
  ``` xml
  <dependency>
      <groupId>...</groupId>
      <artifactId>...</artifactId>
      <version>2.0.0</version>
  </dependency>
  ```

### Version 1.x
  ``` xml
  <dependency>
      <groupId>...</groupId>
      <artifactId>...</artifactId>
      <version>1.0.0</version>
  </dependency>
  ```

File: src/main/java/software/amazon/payloadoffloading/PayloadStorageConfiguration.java [hunk 4]
```diff
@@ -212,4 +182,18 @@ public boolean isAlwaysThroughS3() {
     public void setAlwaysThroughS3(boolean alwaysThroughS3) {
         this.alwaysThroughS3 = alwaysThroughS3;
     }
+
+    public PayloadStorageConfiguration withServiceSideEncryption(ServerSideEncryptionStrategy serverSideEncryptionStrategy) {
```
Comment (adam-aws) [inline src/main/java/software/amazon/payloadoffloading/PayloadStorageConfiguration.java:186]: Can we get some comments on these new methods

File: README.md (hunk 3 above)
Comment (msailes) [reply README.md:36]: done

File: src/main/java/software/amazon/payloadoffloading/PayloadStorageConfiguration.java (hunk 4 above)
Comment (msailes) [reply src/main/java/software/amazon/payloadoffloading/PayloadStorageConfiguration.java]: Done, I've also added a description of the usage in the class docs.

File: src/main/java/software/amazon/payloadoffloading/PayloadStorageConfiguration.java [hunk 5]
```diff
@@ -212,4 +201,38 @@ public boolean isAlwaysThroughS3() {
...
         this.alwaysThroughS3 = alwaysThroughS3;
     }
+
+    /**
+     * Sets which method of server side encryption should be used, if required.
+     *
+     * This is optional, it is set only when you want to configure S3 server side encryption with KMS.
+     *
+     * @param serverSideEncryptionStrategy The method of encryption required for S3 server side encryption with KMS.
+     * @return
```
Comment (eddy-aws) [inline src/main/java/software/amazon/payloadoffloading/PayloadStorageConfiguration.java:211]: Can we complete the `@return` info

File: src/main/java/software/amazon/payloadoffloading/ServerSideEncryptionFactory.java [hunk 6]
```diff
@@ -0,0 +1,11 @@
+package software.amazon.payloadoffloading;
+
+public class ServerSideEncryptionFactory {
```
Comment (eddy-aws) [inline src/main/java/software/amazon/payloadoffloading/ServerSideEncryptionFactory.java:3]: Can we add unit tests for the newly added `SseStrategy` implementations?

File: src/main/java/software/amazon/payloadoffloading/PayloadStorageConfiguration.java (hunk 5 above)
Comment (msailes) [reply src/main/java/software/amazon/payloadoffloading/PayloadStorageConfiguration.java]: done

File: src/main/java/software/amazon/payloadoffloading/ServerSideEncryptionFactory.java (hunk 6 above)
Comment (msailes) [reply src/main/java/software/amazon/payloadoffloading/ServerSideEncryptionFactory.java:3]: done

**Pull request merged**